src/Microsoft.Extensions.Hosting.Abstractions/IHostedService.cs

```diff
@@ -1,22 +1,20 @@
 ﻿// Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
-namespace Microsoft.AspNetCore.Hosting
+namespace Microsoft.Extensions.Hosting
 {
     /// <summary>
     /// Defines methods for objects that are managed by the host.
     /// </summary>
     public interface IHostedService
     {
         /// <summary>
-        /// Triggered when the application host has fully started and the server is waiting
-        /// for requests.
+        /// Triggered when the application host has fully started.
         /// </summary>
         void Start();
```

Comment:
I'm using something very hacky in one of my projects. But maybe it's an acceptable hack for the benefits we gain? 😉 

```csharp
public interface IHostBuilder<THostBuilder>
{
    THostBuilder Instance { get; }
}

public interface IWebHostBuilder : IHostBuilder<IWebHostBuilder>
{
}

public class WebHostBuilder : IWebHostBuilder
{
    public IWebHostBuilder Instance => this;
}

public static class LoggerHostBuilderExtensions
{
    // Only the low-level HostBuilder extensions (logging, DI, configuration) need to use this ugly code.

    public static THostBuilder UseSerilog<THostBuilder>(this IHostBuilder<THostBuilder> builder)
        where THostBuilder : IHostBuilder<THostBuilder>
    {
        // TODO: register Serilog in e.g. builder.Services

        return builder.Instance;
    }
}

public static class WebHostBuilderExtensions
{
    // Extensions for concrete host builders can use the simple code.

    public static IWebHostBuilder UseKestrel(this IWebHostBuilder builder)
    {
        return builder;
    }
}

class Program
{
    static void Main(string[] args)
    {
        // Compiles :)
        new WebHostBuilder()
            .UseSerilog()
            .UseKestrel();
    }
}```